synapse/rest/client/register.py

```diff
@@ -484,9 +484,6 @@ async def on_POST(self, request: SynapseRequest) -> Tuple[int, JsonDict]:
                     "Appservice token must be provided when using a type of m.login.application_service",
                 )
 
-            # Verify the AS
-            self.auth.get_appservice_by_req(request)
```

Comment:
I kinda want to get rid of that method, as it is only used in ~three places, and since we're checking if it's a valid app service request right after in `_do_appservice_registration`, it's not really useful to keep it